File: AndroidStudioProjects/intents/app/src/main/java/com/example/intents/Main2Activity.java
package com.example.intents;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class Main2Activity extends AppCompatActivity {

    TextView t1;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        t1 = findViewById(R.id.textView);
        Intent i = getIntent();
        int result = i.getIntExtra("answer",0);
        t1.setText(String.valueOf(result));

    }
}

File: AndroidStudioProjects/kotlinintents/app/src/main/java/com/example/kotlinintents/Main2Activity.kt
package com.example.kotlinintents

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main2.*

class Main2Activity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)

        val result = intent.getIntExtra(KEY_1,0)
        textView.text = result.toString()
    }
}
